ngraph/core/include/openvino/core/node.hpp

```diff
@@ -305,75 +304,139 @@ class OPENVINO_API Node : public std::enable_shared_from_this<Node> {
 
     /// Returns the output of the default output, or throws if there is none
+    OPENVINO_DEPRECATED("The method was deprecated and will be removed soon!")
     virtual size_t get_default_output_index() const;
     /// Throws no default
+    OPENVINO_DEPRECATED("The method was deprecated and will be removed soon!")
     size_t no_default_index() const;
 
     /// Checks that there is exactly one output and returns its shape
-    // TODO: deprecate in favor of node->get_output_shape(0) with a suitable check in the
-    // calling code, or updates to the calling code if it is making an invalid assumption of
-    // only one output.
+    OPENVINO_DEPRECATED("The method was deprecated! Please use output_shape() instead.")
     const StaticShape& get_shape() const;
 
     /// Returns the tensor for output or input i
+    OPENVINO_DEPRECATED("The method was deprecated! Please use output_tensor() instead.")
     descriptor::Tensor& get_output_tensor(size_t i) const;
-    descriptor::Tensor& get_input_tensor(size_t i) const;
 
     /// Returns the tensor name for output i
     OPENVINO_DEPRECATED("The tensor name was deprecated. Use get_output_tensor(i).get_names() instead.")
     const std::string& get_output_tensor_name(size_t i) const;
 
+    OPENVINO_DEPRECATED("The tensor name was deprecated. Use output_tensor(i).get_target_inputs() instead.")
     std::set<Input<Node>> get_output_target_inputs(size_t i) const;
 
+    /// ************* Outputs ***************
+
+    /// ************* Inputs  ***************
+    /// \return A vector containing a handle for each of this node's inputs, in order.
+    std::vector<Input<Node>> inputs();
+
+    /// \return A vector containing a handle for each of this node's inputs, in order.
+    std::vector<Input<const Node>> inputs() const;
+
+    /// \return A vector containing the values for each input
+    std::vector<Output<Node>> input_values() const;
+
+    /// \return A handle to the `input_index`th input of this node.
+    /// \throw std::out_of_range if the node does not have at least `input_index+1` inputs.
+    Input<Node> input(size_t input_index);
+
+    /// \return A handle to the `input_index`th input of this node.
+    /// \throw std::out_of_range if the node does not have at least `input_index+1` inputs.
+    Input<const Node> input(size_t input_index) const;
+
+    Output<Node> input_value(size_t input_index) const;
+
+    size_t input_size() const;
+
+    const Shape& input_shape(size_t i) const;
+    const element::Type& input_element_type(size_t i) const;
+    descriptor::Tensor& input_tensor(size_t i) const;
+    Node* input_node_ptr(size_t index) const;
+    std::shared_ptr<Node> input_node_shared_ptr(size_t index) const;
+    Output<Node> input_source_output(size_t i) const;
+    /// Returns the tensor for nput i
+    OPENVINO_DEPRECATED("The method was deprecated! Please use input_tensor() instead.")
+    descriptor::Tensor& get_input_tensor(size_t i) const;
```

Comment:
it looks too much changes which don't bring any benefits

Reply:
It allows to align the function names.
From my point of view it is strange to have a mix of methods:
```
outputs();
output();
inputs();
input();
input_shape();
output_shape();

// and
get_input_element_type()
get_input_tensor()
// etc.
```

Reply:
> It allows to align the function names.

Which one? Please, provide a description what we are trying to do

Reply:
Let's discuss it offline